pyspider/scheduler/scheduler.py

```diff
@@ -574,7 +574,7 @@ def on_old_request(self, task, old_task):
         schedule_age = _schedule.get('age', self.default_schedule['age'])
         if _schedule.get('itag') and _schedule['itag'] != old_schedule.get('itag'):
             restart = True
-        elif schedule_age >= 0 and schedule_age + (old_task['lastcrawltime'] or 0) < now:
+        elif 'lastcrawltime' in old_task and schedule_age >= 0 and schedule_age + (old_task['lastcrawltime'] or 0) < now:
```

Comment:
yes. mongodb.
Seems its because of journals?

2015-04-22 6:43 GMT+04:00 Roy Binux <notifications@github.com>:

> In pyspider/scheduler/scheduler.py
> <https://github.com/binux/pyspider/pull/191#discussion_r28840853>:
>
> > @@ -574,7 +574,7 @@ def on_old_request(self, task, old_task):
> >          schedule_age = _schedule.get('age', self.default_schedule['age'])
> >          if _schedule.get('itag') and _schedule['itag'] != old_schedule.get('itag'):
> >              restart = True
> > -        elif schedule_age >= 0 and schedule_age + (old_task['lastcrawltime'] or 0) < now:
> > +        elif 'lastcrawltime' in old_task and schedule_age >= 0 and schedule_age + (old_task['lastcrawltime'] or 0) < now:
>
> elif schedule_age >= 0 and schedule_age + old_task.get('lastcrawltime', 0)
> < now:
> could be better.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/binux/pyspider/pull/191/files#r28840853>.
>